os/arch/arm/src/amebad/amebad_reboot_reason.c

```diff
@@ -56,26 +56,50 @@
 #include <tinyara/config.h>
 
 #include <tinyara/reboot_reason.h>
+#include "ameba_soc.h"
 /****************************************************************************
  * Public functions
  ****************************************************************************/
 #ifdef CONFIG_SYSTEM_REBOOT_REASON
+
+static reboot_reason_code_t backup_reg;
+
 void up_reboot_reason_init(void)
```

Comment:
In this case, we'd better call the `up_reboot_reason_init()` when board bring up.
Please add it.